Master

DavidARivkin opened the pull request:
Peak Investigator primary integration code changes implemented.
This needs to be Tested to ensure compatibility with existing and expected functionality.

DavidARivkin committed:
PI Integration

Veritomyx Peak Investigator codes added.

DavidARivkin committed:
Veritomyx Integration

Integrating Veritomyx functions with the core code.

TDI-tenderholt committed:
POM: update to use OpenSftp and Jtar

* OpenSftp is not in the public Maven repo, so it needs to be downloaded
  from https://code.google.com/p/opensftp/downloads/list.

* The JARs need to be installed to the local maven repo with following
  commands:

  mvn install:install-file -Dfile=opensftp-api-0.3.jar
  -DgroupId=net.sf \
  -DartifactId=opensftp-api -Dversion=0.3 -Dpackaging=jar

  mvn install:install-file -Dfile=opensftp-impl-0.3.0.jar \
  -DgroupId=net.sf -DartifactId=opensftp-impl -Dversion=0.3.0 \
  -Dpackaging=jar

* Unsure if the other JARs distributed in the lib directory of the
  OpenSftp download need to also be added to pom.

TDI-tenderholt committed:
POM: add org.jasypt as requested by David

DavidARivkin committed:
Veritomyx Peak Investigator Integration

Integrating Veritomyx Peak Investigator code.

DavidARivkin committed:
Merge remote-tracking branch 'TDI-tenderholt/pom_updates'

DavidARivkin committed:
Peak Investigator Integration

Integrating more of Peak Investigator Codes

DavidARivkin committed:
Integration of Peak Investigator

Integration of Peak Investigator modules and support files

DavidARivkin committed:
Peak Investigator Integration

Integrating Peak Investigator, particularly DBSearch, with the main
code.

DavidARivkin committed:
Peak Investigator integration

More Peak Investigator integrations for the main code.

TDI-tenderholt commented:
I'm currently getting some build errors, several in KEGGGateway.java. Did you install anything to help with those errors?

TDI-tenderholt committed:
Remove various files that are no longer in upstream MZmine

* No longer a peaklistmethods.identification.dbsearch module
* Changes: to peaklistmethods.identification.formulaprediction module:
  - elements.ElementsRule and elements.ElementsRuleSorterByMass have
    been removed
  - FormulaGenerator has been removed

DavidARivkin committed:
Remove Unneeded files

Removed modules/peaklistmethods/identification/formulaprediction extra
files (from PI Investigator integration) that are not needed in 2.17
base codes.

DavidARivkin committed:
Remove redundant DBSearch

Removed the redundant DBSearch (now called onlinedbsearch) from the PI
code.

DavidARivkin committed:
Up to date POM

Updated POM with all dependencies needed.

TDI-tenderholt commented on src/main/java/net/sf/mzmine/datamodel/PeakStatus.java:
```diff
@@ -0,0 +1,49 @@
+/*
```
What is this file for? I don't see it it in the main MZmine project.

https://github.com/mzmine/mzmine2/tree/master/src/main/java/net/sf/mzmine/datamodel

TDI-tenderholt commented on src/main/java/net/sf/mzmine/datamodel/impl/RemoteJob.java:
```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2013-2015 The Veritomyx
```
Change from `The Veritomyx` to `Veritomyx, Inc.`.

TDI-tenderholt commented on src/main/java/net/sf/mzmine/main/MZmineCore.java:
```diff
@@ -59,6 +60,13 @@
  */
 public final class MZmineCore {
 
+	public static final String MZmineName      = "MZmine PeakInvestigator™ Edition";
```
The `MZmineName` and related variables should probably not make it into the push to the upstream project. Can you look into whether this info can be exposed via `pom.xml` and the Maven build system? I believe you can have a properties file in `src/main/resources` and the following in the POM:

```
        <resources>
                <resource>
                        <directory>src/main/resources</directory>
                        <filtering>true</filtering>
                </resource>
        </resources>
```

TDI-tenderholt commented on src/main/java/net/sf/mzmine/modules/masslistmethods/listexport/help/help.html:
```diff
@@ -0,0 +1,98 @@
+<html>
```
Was this added in the MZminePI code? It doesn't look like anything Dan would have written, but it seems to be help for the `listexport` module that he wrote.

DavidARivkin replied:
Apparently it was.  It is in the Beta code directory.

On Fri, Oct 9, 2015 at 9:47 PM, TDI-tenderholt notifications@github.com
wrote:

> In
> src/main/java/net/sf/mzmine/modules/masslistmethods/listexport/help/help.html
> https://github.com/TDI-tenderholt/mzmine2/pull/1#discussion_r41664838:
> 
> > @@ -0,0 +1,98 @@
> > +<html>
> 
> Was this added in the MZminePI code? It doesn't look like anything Dan
> would have written, but it seems to be help for the listexport module
> that he wrote.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/TDI-tenderholt/mzmine2/pull/1/files#r41664838.

TDI-tenderholt commented on src/main/java/net/sf/mzmine/modules/rawdatamethods/peakpicking/massdetection/Veritomyx/Veritomyx.java:
```diff
@@ -0,0 +1,177 @@
+/*
```
Consider renaming this class from `Veritomyx` to `PeakInvestigator`.

TDI-tenderholt replied:
Also, I think the package should change from `massdetection.Veritomyx` to `massdetection.PeakInvestigator`, and this class should be `PeakInvestigatorDetector`, to better match the naming conventions of other `massdetection` packages (e.g. https://github.com/mzmine/mzmine2/tree/master/src/main/java/net/sf/mzmine/modules/rawdatamethods/peakpicking/massdetection/localmaxima).

TDI-tenderholt commented on src/main/java/net/sf/mzmine/modules/rawdatamethods/peakpicking/massdetection/Veritomyx/help/help.html:
```diff
@@ -0,0 +1,122 @@
+
+<body>
+
+<h2>MZmine PeakInvestigator™ Edition 2.10.37</h2>
```
Update this to 2.17.

TDI-tenderholt commented on src/main/java/net/sf/mzmine/modules/rawdatamethods/peakpicking/massdetection/exactmass/ExactMassDetector.java:
```diff
@@ -37,50 +38,57 @@
+    	if (!selected)	// only process selected scans
+    		return null;
+	
+		double noiseLevel = parameters.getParameter(
```
Fix the indentation so that the bulk of this function doesn't appear to fall under the `if(!selected)` statement.

You should probably also make sure you load and use the proper formatting style, which is available in the `mzmine-eclipse-code-formatter.xml` file. Perhaps do a single commit that is only formatting updates.

TDI-tenderholt commented on src/main/java/net/sf/mzmine/modules/rawdatamethods/peakpicking/massdetection/localmaxima/LocalMaxMassDetector.java:
```diff
@@ -33,62 +34,70 @@
  */
 public class LocalMaxMassDetector implements MassDetector {
 
-    public DataPoint[] getMassValues(Scan scan, ParameterSet parameters) {
```
It looks like these changes are only formatting. Is that the case?

TDI-tenderholt commented on src/main/java/net/sf/mzmine/modules/visualization/spectra/PlotMode.java:
```diff
@@ -0,0 +1,24 @@
+/*
```
This file doesn't appear to be in the upstream project. https://github.com/mzmine/mzmine2/tree/master/src/main/java/net/sf/mzmine/modules/visualization/spectra

DavidARivkin replied:
That is correct.  It is not there and is needed for one of the
visualizations that PI added as it contains the defines that are not the
same as what MZmine uses.  I mentioned this in an email to you.

David

On Fri, Oct 9, 2015 at 9:59 PM, TDI-tenderholt notifications@github.com
wrote:

> In src/main/java/net/sf/mzmine/modules/visualization/spectra/PlotMode.java
> https://github.com/TDI-tenderholt/mzmine2/pull/1#discussion_r41666117:
> 
> > @@ -0,0 +1,24 @@
> > +/*
> 
> This file doesn't appear to be in the upstream project.
> https://github.com/mzmine/mzmine2/tree/master/src/main/java/net/sf/mzmine/modules/visualization/spectra
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/TDI-tenderholt/mzmine2/pull/1/files#r41666117.

TDI-tenderholt commented on src/main/java/net/sf/mzmine/parameters/parametertypes/PasswordParameter.java:
```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 20013-2014 The MZminePI Development Team
```
Change copyright to Veritomyx, Inc.

TDI-tenderholt commented on src/main/java/net/sf/mzmine/parameters/parametertypes/tolerances/MZToleranceParameter.java:
```diff
@@ -128,8 +128,8 @@ public boolean checkValue(Collection<String> errorMessages) {
 	    return false;
```
Fix formatting so this diff doesn't appear.

TDI-tenderholt commented on src/main/java/veritomyxSaaS/FileChecksum.java:
```diff
@@ -0,0 +1,262 @@
+package veritomyxSaaS;
```
Change package to `com.veritomyx` to better match standard Java conventions.

DavidARivkin committed:
Peak Investigator related cleanup

Updated organization and naming of P.I. classes, and copyright data, and
help file information.

TDI-tenderholt committed:
MZmineCore: remove short name since it seems pointless

* No benefit to using MZminePI.logging.properties vs.
  logging.properties.

TDI-tenderholt committed:
POM: use Maven buildNumber to insert compile info into manifest

DavidARivkin committed:
PeakInvestigatorSaaS

Changed VeritomyxSaas to PeakInvestigatorSaaS

TDI-tenderholt committed:
MZmineCore: attempt to read name/version info from the Manifest

DavidARivkin committed:
Merge pull request #3 from TDI-tenderholt/version_cleanup

Version cleanup

DavidARivkin committed:
Merge remote-tracking branch 'refs/remotes/TDI-tenderholt/compile_fixes'

DavidARivkin committed:
Merge remote-tracking branch 'refs/remotes/TDI-tenderholt/version_cleanup'

DavidARivkin committed:
MZmineName via POM.xml

Set the MZmineName variable to be the project name from the POM.xml
file, allowing for custom product names.

DavidARivkin committed:
Merge branch 'master' of https://github.com/DavidARivkin/mzmine2

DavidARivkin committed:
Initial "Gamma" version support

Initial changes for the 2.5 "Gamma" version of the Peak Investigator
API.

DavidARivkin committed:
Update for "Gamma" 2.5 Peak Investigator

Added functional and UI support for new features and issues with version
2.5 API "Gamma" of Peak Investigator SaaS

TDI-tenderholt commented:
Can you create a new branch for Beta work, reset its HEAD back to ```827df2f```, and create a new PR so that I can pull in just the updates to MZmine 2.17? We should make sure all of the features work there before moving really starting Gamma work.

DavidARivkin committed:
Revert "Update for "Gamma" 2.5 Peak Investigator"

This reverts commit 4aecc319bb9798b44841a2e86af14d3e73e6a0c1.

DavidARivkin committed:
Revert "Initial "Gamma" version support"

This reverts commit b1e8e62bd51a1265b09c6219183c511da09fd085.

DavidARivkin committed:
Fixes to integration of PI with 2.17

Fixed needed to support async needs of Peak Investigator via
MassDetectionTask

TDI-tenderholt committed:
Allow Scan export dialog to choose output directory

DavidARivkin committed:
Merge pull request #4 from TDI-tenderholt/list_export_fix

Allow Scan export dialog to choose output directory

TDI-tenderholt committed:
Compile fixes: move veritomyx.gamma to veritomyx/gamma

TDI-tenderholt committed:
MassDetector.getMassValues(): remove selected parameter

* Now that MZmine has a mechanism for choosing scans, the hack Dan made
  to allow a subset of scans to be chosen is no longer needed.

TDI-tenderholt committed:
POM + libs: updates for runtime dependencies of OpenSftp

DavidARivkin committed:
Merge pull request #5 from TDI-tenderholt/updates

Various fixes & updates

TDI-tenderholt committed:
MassDetectionTask: minor bug fix when tar'ing up files.

TDI-tenderholt committed:
POM: add/update some more OpenSftp dependencies

TDI-tenderholt committed:
RawDataTreeModel: update to insert remote jobs

TDI-tenderholt committed:
PeakInvestigatorSaaS: don't try removing (non)existing files

* An exception was thrown due to a file not existing, and it was unclear
  how to avoid this with the OpenSftp API. Since it is unlikely we would
  ever re-upload a file with the exact same name, this seems uncessary
  and removing it provides the necessary fix.

TDI-tenderholt committed:
Add support for saving/loading jobs that are currently running

TDI-tenderholt committed:
ProjectTreeHandler: add support for PeakInvestigator jobs

* Still doesn't fetch the job though...

DavidARivkin committed:
Merge pull request #6 from TDI-tenderholt/updates

More fixes and updates

TDI-tenderholt committed:
MassDetectionParameters: fix to allow downloading/processing results

* With the new RawDataFilesSelection, need to explicitly set that
  SPECIFIC_FILES are desired when calling setJobParams().

TDI-tenderholt committed:
ProjectTreeMouseHandler: make right clicks in tree behave as expected

DavidARivkin committed:
Merge pull request #7 from TDI-tenderholt/updates

More updates

TDI-tenderholt committed:
ListExportModule: bug fix when exporting data as multiple text files

TDI-tenderholt committed:
MassListParameter: allows the specification of whether its required

* In some cases (e.g. List export), it may be desirable to not require
  the mass list parameter. This commit add a new construction that takes
  a boolean indicating if its required, and checkValue() first checks if
  it's required.

TDI-tenderholt committed:
ListExport: improvements to behavior

* Don't require the MassListParameter
* Show dialog if nothing was chosen to be exported.

DavidARivkin committed:
Merge pull request #8 from TDI-tenderholt/list_export_fix

More fixes to the exporting of scans

DavidARivkin committed:
GirHub Lost Updates

Somehow GitHub lost all these updates.

DavidARivkin committed:
Updated message for better English

The waiting for PREP message is updated for better clarify and use of
English.

DavidARivkin committed:
PREP timeline and check time

Implementation of adjustable timeout and check rate for PREP

DavidARivkin committed:
RUN and STATUS

Run submissions and STATUS (as far as Running) are not working.
